gapic/schema/api.py

```diff
@@ -560,6 +561,116 @@ def mixin_http_options(self):
             res[s] = [rule for rule in opt_gen if rule]
         return res
 
+    @cached_property
+    def all_methods(self) -> Mapping[str, MethodDescriptorProto]:
+        """Return a map of all methods for the API.
+
+        Return:
+            Mapping[str, MethodDescriptorProto]: A mapping of MethodDescriptorProto
+                values for the API.
+        """
+        return {
+            f"{service_key}.{method_key}": method_value
+            for service_key, service_value in self.services.items()
+            for method_key, method_value in service_value.methods.items()
+        }
+
+    def check_method_settings_validity(
+        self, service_method_settings: Sequence[client_pb2.MethodSettings]
+    ) -> None:
+        """
+        Checks each `google.api.client.MethodSettings` for validity. If
+        `google.api.client.MethodSettings.auto_populated_fields`
+        is set, verify each field against the criteria of AIP-4235
+        (https://google.aip.dev/client-libraries/4235). All the configurations 
+        below must be true:
+
+        - The field must be of type string
+        - The field must be at the top-level of the request message
+        - The RPC must be a unary RPC (i.e. streaming RPCs are not supported)
+        - The field must not be annotated with google.api.field_behavior = REQUIRED.
+        - The field must be annotated with google.api.field_info.format = UUID4.
+        - The field presence requirements in AIP-4235 are checked at run time.
+
+        Args:
+            service_method_settings (Sequence[client_pb2.MethodSettings]): Method settings to be used
+                when generating API methods.
+        Return:
+            None
+        Raises:
+            ValueError: if fields in `method_settings.auto_populated_fields` cannot be
+                automatically populated.
+        """
+
+        all_errors: str = ""
+        invalid_selectors = []
+        for method_settings in service_method_settings:
+            method_descriptor = self.all_methods.get(method_settings.selector)
+            if not method_descriptor:
+                all_errors += f"Selector {method_settings.selector} is not a valid method in this API. "
+                invalid_selectors.append(method_settings.selector)
+                continue
+            if method_settings.auto_populated_fields:
+                # The method must not be a streaming method.
+                if method_descriptor.client_streaming or method_descriptor.server_streaming:
+                    all_errors += f"Selector {method_settings.selector} is a streaming method. `auto_populated_fields` are only supported for unary methods. "
```

Comment:
```suggestion
                    all_errors += f"Selector {method_settings.selector} is a streaming method. Only unary methods support the `auto_populated_fields` setting. "
```

(Just so we can capitalize the first word of the second sentence ;-) )

Reply:
Fixed in 5a2d33b8b160a246945e3752e145878581a351b1